### PR TITLE
HTTP Beacon API: `/eth/v1/node/syncing`

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

HTTP handler for https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Node/getSyncingStatus

Benchmark shows less than 1ms during regular node operation. During initial sync the request can take several seconds because it's blocked waiting for a block batch to process. This will be resolved in another PR.

### CODE REVIEW

**Review by saolyn — APPROVED** [review]
LGTM

File: beacon-chain/rpc/eth/node/handlers.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	ctx, span := trace.StartSpan(r.Context(), "node.GetSyncStatus")
+	defer span.End()
+
+	start := time.Now()
+	isOptimistic, err := s.OptimisticModeFetcher.IsOptimistic(ctx)
+	if err != nil {
+		http2.HandleError(w, "Could not check optimistic status: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	log.Warnf("IsOptimistic: %s", time.Since(start))
```
Comment (james-prysm) [inline beacon-chain/rpc/eth/node/handlers.go:25]: Are these logs supposed to be there?

File: beacon-chain/rpc/eth/node/handlers.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+	headSlot := s.HeadFetcher.HeadSlot()
```
Comment (james-prysm) [inline beacon-chain/rpc/eth/node/handlers.go:27]: hmm is the headslot fetcher always supposed to be before the syncing and execution client checks?

is this something we should group together for safer calls in the future?

File: beacon-chain/rpc/eth/node/handlers.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	isOptimistic, err := s.OptimisticModeFetcher.IsOptimistic(ctx)
+	if err != nil {
+		http2.HandleError(w, "Could not check optimistic status: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	headSlot := s.HeadFetcher.HeadSlot()
+	response := &SyncStatusResponse{
+		Data: &SyncStatusResponseData{
+			HeadSlot:     strconv.FormatUint(uint64(headSlot), 10),
```
Comment (james-prysm) [inline beacon-chain/rpc/eth/node/handlers.go:26]: is it preferred to use strconv.FormatUint here?

i have used fmt.Sprintf("%d", myuint) for all my string printing on numbers

Comment (james-prysm) [reply beacon-chain/rpc/eth/node/handlers.go:26]: we decided to keep it as is

**Review by james-prysm — APPROVED** [review]

**Comment by saolyn** [comment]
Part of #12635

**Pull request merged**